### PR TITLE
DOC: Add docs for gcc/clang build

### PR DESCRIPTION
**Pull request by Smit-create**
#36

### CODE REVIEW

**Comment by Smit-create** [comment]
@rgommers Please have a look at the added docs. Once, this looks I will open a PR in main repo.

File: doc/source/dev/contributor/meson.rst [hunk 1]
```diff
@@ -137,6 +137,42 @@ To run the tests for a particular submodule(let's say ``optimize``), you can use
...
+=================================
+
+As discussed, Meson is fully out-of-place, so different builds will not interfere
+with each other. For example, let us build using GCC and Clang.
+
+1. Build with GCC::
+
+    export CC=gcc
+    meson setup build-gcc --prefix=$PWD/install-gcc
+    ninja -C build-gcc
```
Comment (rgommers) [inline doc/source/dev/contributor/meson.rst:150]: Why not use `python dev.py` here? We no longer need the separate `setup`/`build`/`install` steps.

File: doc/source/dev/contributor/meson.rst [hunk 2]
```diff
@@ -137,6 +137,42 @@ To run the tests for a particular submodule(let's say ``optimize``), you can use
...
+
+    meson install -C build-gcc
+
+It will then install to ``$PWD/install-gcc/lib/python3.x/site-packages/scipy``.
+
+2. Build with Clang::
+
+    export CC=clang
+    meson setup build-clang --prefix=$PWD/install-clang
+    ninja -C build-clang
```
Comment (rgommers) [inline doc/source/dev/contributor/meson.rst:163]: Note that you also need to set `CXX`. So I would do all this in one go:
```
CC=clang CXX=clang++ python dev.py --build-only --build-dir=build-clang
```
And then below it note that Meson will remember the compiler selection for the `build-clang` dir and it cannot be changed, so each future invocation of `python dev.py --build-dir=build-clang` will automatically use Clang.

And then to run tests:
```
python dev.py --build-dir build-clang -s linalg
```

**Review by rgommers — COMMENTED** [review]
Hi @Smit-create, here are some suggested changes

**Comment by Smit-create** [comment]
> Hi @Smit-create, here are some suggested changes

Thanks! I updated it.

**Pull request closed without merging**